tests/kubearmor_config/kubearmor_config_test.go

```diff
@@ -0,0 +1,279 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Authors of KubeArmor
+
+package kubearmor_config
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/kubearmor/KubeArmor/protobuf"
+	. "github.com/kubearmor/KubeArmor/tests/util"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = BeforeSuite(func() {
+	// install wordpress-mysql app
+	err := K8sApply([]string{"manifests/ubuntu-deployment.yaml"})
+	Expect(err).To(BeNil())
+
+	// delete all KSPs
+	KspDeleteAll()
+
+	// enable kubearmor port forwarding
+	err = KubearmorPortForward()
+	Expect(err).To(BeNil())
+})
+
+var _ = AfterSuite(func() {
+	KubearmorPortForwardStop()
+	CreateKAConfigMap("block", "block", "block")
+	DeleteKAConfigMap()
+})
+
+func getUnannotatedPod(name string, ant string) string {
+	pods, err := K8sGetPods(name, "unannotated", []string{ant}, 60)
+	Expect(err).To(BeNil())
+	Expect(len(pods)).To(Equal(1))
+	return pods[0]
+}
+
+func getFullyAnnotatedPod(name string, ant string) string {
+	pods, err := K8sGetPods(name, "fullyannotated", []string{ant}, 60)
+	Expect(err).To(BeNil())
+	Expect(len(pods)).To(Equal(1))
+	return pods[0]
+}
+
+func getPartialyAnnotatedPod(name string, ant string) string {
+	pods, err := K8sGetPods(name, "partialyannotated", []string{ant}, 60)
+	Expect(err).To(BeNil())
+	Expect(len(pods)).To(Equal(1))
+	return pods[0]
+}
+
+func getKarmorLog(alerts []protobuf.Alert, target protobuf.Alert) bool {
+	for _, alert := range alerts {
+		if alert.PolicyName != target.PolicyName {
+			continue
+		}
+		if alert.Action != target.Action {
+			continue
+		}
+		if alert.Result != target.Result {
+			continue
+		}
+		if alert.NamespaceName != target.NamespaceName {
+			continue
+		}
+		return true
+	}
+	return false
+}
+
+var _ = Describe("KubeArmor-Config", func() {
+	var unannotated string
+	var partialyAnnotated string
+	var fullyAnnotated string
+	BeforeEach(func() {
+		unannotated = getUnannotatedPod("unannotated-", "container.apparmor.security.beta.kubernetes.io/ubuntu-1: localhost/kubearmor-unannotated-unannotated-deployment-ubuntu-1")
+		partialyAnnotated = getPartialyAnnotatedPod("partialyannotated-", "container.apparmor.security.beta.kubernetes.io/ubuntu-1: localhost/kubearmor-partialyannotated-partialyannotated-deployment-ubuntu-1")
+		fullyAnnotated = getFullyAnnotatedPod("fullyannotated-", "container.apparmor.security.beta.kubernetes.io/ubuntu-1: localhost/kubearmor-fullyannotated-fullyannotated-deployment-ubuntu-1")
+		CreateKAConfigMap("audit", "audit", "audit")
+	})
+
+	AfterEach(func() {
+		KarmorLogStop()
+		KspDeleteAll()
+		time.Sleep(5 * time.Second)
+	})
+
+	Describe("Unannotated", func() {
+
+		It("default posture will be set to global posture configs", Label("unannotated"), func() {
+			// apply a allow based policy
+			err := K8sApply([]string{"manifests/ksp-unannotated-allow.yaml"})
+			Expect(err).To(BeNil())
+
+			// wait for policy
+			time.Sleep(5 * time.Second)
```

Comment:
we dont need sleep after apply anymore, apply will sleep for 5 seconds before returning

Reply:
yes the tests are need to be refactored now as #778 merged. I'm on on it